### PR TITLE
Improve feedback when saving locations, add toast notifications

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR improves the user feedback from setting the home or work locations, from the map and the places editor, notably with toast notifications or alerting logged out users they need to log in. Some other confirmation alert messages are also converted into toast notifications.

Under the hood, the added `react-hot-toast` package improves the "politeness" of the confirmation messages shown.

## PR Checklist
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/util/user.js [hunk 1]
```diff
@@ -138,7 +139,7 @@ export function convertToLegacyLocation(place) {
     lon,
     // HACK: If a place name and address are provided, put the address in parentheses
     // to mimic the existing LocationField behavior for "work" and "home".
-    // TODO: use addInParentheses from location-field (requires passing an intl contexty).
+    // TODO: use addInParentheses from location-field (requires passing an intl context).
```
Comment (amy-corson-ibigroup) [inline lib/util/user.js:142]: 👍

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Tested on mobile and AT. 

It would be great to move all `window.alerts` over to Toast, as this does make the user alerts a little inconsistent, but I think this is a great place to start!!

File: i18n/en-US.yml [hunk 2]
```diff
@@ -52,6 +52,8 @@ actions:
     emailVerificationResent: The email verification message has been resent.
     genericError: "An error was encountered: {err}"
     itineraryExistenceCheckFailed: Error checking whether your selected trip is possible.
+    mustBeLoggedInToSavePlace: Please log in to save locations.
```
Comment (AdrianaCeric) [inline i18n/en-US.yml:55]: [Testing] How is it possible to save locations without logging in?

File: i18n/en-US.yml [hunk 3]
```diff
@@ -52,6 +52,8 @@ actions:
...
+    placeRemembered: "Your {placeType} has been set to: {address}."
```
Comment (AdrianaCeric) [inline i18n/en-US.yml:56]: `placeType` is either "custom" or "dining" which results in the alert becoming either:
- "Your custom has been set to: Sesame Street."
- "Your dining has been set to: Sesame Street."

Adding something like "place type" would help make this alert more clear
```suggestion
    placeRemembered: "Your {placeType} place type has been set to: {address}."
```

**Review by AdrianaCeric — CHANGES_REQUESTED** [review]
I think Toast is a cool package, good work! Just one comment -- also not sure how to test the login problem

File: i18n/en-US.yml (hunk 3 above)
Comment (binh-dam-ibigroup) [reply i18n/en-US.yml]: Good point! I changed the confirmation popup in https://github.com/opentripplanner/otp-react-redux/pull/992/commits/e1c64536b1f836a7597349d1e54636fe62043861 so it shows the correct place name, and did away with showing the address as there can be other settings for places later.

**Review by AdrianaCeric — APPROVED** [review]
Looks good with the change!

**Pull request merged**